story_types.py

```diff
@@ -0,0 +1,282 @@
+"""
+Story Types and SubTypes Implementation
+
+This module implements the seven classical story types and their subtypes
+as described in narrative theory.
+"""
+
+from typing import List, Optional
+from dataclasses import dataclass, field
+
+
+@dataclass
+class StorySubType:
+    """Represents a specific subtype within a story type."""
+    name: str
+    description: str
+    examples: List[str] = field(default_factory=list)
+    
+    def __str__(self) -> str:
+        return f"{self.name}: {self.description}"
+
+
+@dataclass
+class StoryType:
+    """Base class for all story types."""
+    name: str
+    description: str
+    subtypes: List[StorySubType] = field(default_factory=list)
+    narrative_elements: Optional[str] = None
```

Comment:
Convert narrative_elements, key_theme, emotional_arc and common_elements in into a single field called "characteristics"